web/src/features/moreCast2/components/MoreCast2Field.tsx

```diff
@@ -0,0 +1,193 @@
+import { GridColDef, GridValueFormatterParams } from '@mui/x-data-grid'
+import { DateTime } from 'luxon'
+import { GridNumberRenderer } from 'features/moreCast2/components/DataGridNumberRenderer'
+
+export type Morecast2Field = 'stationName' | 'forDate' | 'temp' | 'rh' | 'windDirection' | 'windSpeed' | 'precip'
+
+export interface ForecastField {
+  field: Morecast2Field
+  headerName: string
+  type: 'number' | 'string'
+  precision: 0 | 1
+  generateColDef: () => GridColDef
+}
+
+export class StationForecastField implements ForecastField {
+  private static instance: StationForecastField
+
+  readonly field = 'stationName'
+  readonly headerName = 'Station'
+  readonly type = 'string'
+  readonly precision = 0
+  private constructor() {
+    /* no op */
+  }
+  public generateColDef = () => {
+    return { field: this.field, flex: 1, headerName: this.headerName, maxWidth: 200 }
+  }
+
+  public static getInstance(): StationForecastField {
+    if (!StationForecastField.instance) {
+      StationForecastField.instance = new StationForecastField()
+    }
+
+    return StationForecastField.instance
+  }
+}
+
+export class DateForecastField implements ForecastField {
+  private static instance: DateForecastField
+
+  readonly field = 'forDate'
+  readonly headerName = 'Date'
+  readonly type = 'string'
+  readonly precision = 0
```

Comment:
nit-picking suggestion - why not remove the `precision` attribute on the `DateForecastField` and `StationForecastField ` components since it doesn't make sense to have precision on a string, and remove the `precision` attribute from the ForecastField component, and only have it on the components that store numeric values?

Reply:
Because it makes the code more complex lol, it's just ignored.